src/ForceHighPoly.Core/ForceHighPoly.Hooks.cs

```diff
@@ -1,36 +1,224 @@
+                        var currentcloth = clothParts[j];
+                        var id = currentcloth.id;
+                        var work = __instance.lstCtrl.GetCategoryInfo((ChaListDefine.CategoryNo)category);
+                        if (!work.TryGetValue(id, out var lib))
+                        {
+                            continue;
+                        }
+                        else if (category == 105 || category == 107)
+                        {
+                            var infoInt = lib.GetInfoInt(ChaListDefine.KeyType.Sex);
+                            if (__instance.sex == 0 && infoInt == 3 || __instance.sex == 1 && infoInt == 2)
+                            {
+                                if (id != 0)
+                                {
+                                    work.TryGetValue(0, out lib);
+                                }
+                                if (lib == null)
+                                {
+                                    continue;
+                                }
+                            }
+                        }
+                        var highAssetName = lib.GetInfo(ChaListDefine.KeyType.MainData);
+                        if (string.Empty == highAssetName)
+                        {
+                            continue;
+                        }
+                        var manifestName = lib.GetInfo(ChaListDefine.KeyType.MainManifest);
+                        var assetBundleName = lib.GetInfo(ChaListDefine.KeyType.MainAB);
+#if KK
+                        Singleton<Manager.Character>.Instance.AddLoadAssetBundle(assetBundleName, manifestName);
+#elif KKS
+                        Manager.Character.AddLoadAssetBundle(assetBundleName, manifestName);
 #endif
+                        if (!CommonLib.LoadAsset<UnityEngine.GameObject>(assetBundleName, lib.GetInfo(ChaListDefine.KeyType.MainData) + "_low", false, manifestName) && CommonLib.LoadAsset<UnityEngine.GameObject>(assetBundleName, highAssetName, false, manifestName))
                         {
-                            assetName = assetName.Substring(0, assetName.Length - 4);
+                            __instance.hiPoly = true;
+                            return;
                         }
                     }
                 }
             }
+
+            /// <summary>
+            /// Might not be neccessary because of the first line of the above method, put probably also cheaper than the previous method
+            /// Equivilant to last method results of patching AssetBundleManager and trimming _low
+            /// </summary>
+            /// <param name="__result"></param>
+            [HarmonyPostfix, HarmonyPatch(typeof(ChaInfo), nameof(ChaInfo.hiPoly), MethodType.Getter)]
+            private static void HiPolyPostfix(ref bool __result)
+            {
+                __result = __result || Enabled.Value;
+            }
+            #region Transpile code that works, but animation locks
```

Comment:
Probably should be removed before merging. If it's possibly going to be useful in the future it's better to keep it in a branch or a discussion.